### ISSUE TITLE
[camelcase] False positive when using leading underscore when destructuring with computed object property name.

### ISSUE THREAD

**Issue by apples** [eslint/eslint#13022]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.16.1
* **npm Version:** v6.14.2

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 2020
    },
    "rules": {
        "camelcase": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const obj = {};

const { ['foo']: _foo } = obj;
```

```bash
eslint ./main.js
```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

Error:

```
<redacted>\main.js
  4:18  error  Identifier '_foo' is not in camel case  camelcase

✖ 1 problem (1 error, 0 warnings)
```

Seems to be caused by the computed object property name in the `{ ['foo']: _foo }` destructuring syntax. There is no error if it is changed to `{ foo: _foo }`.

**Are you willing to submit a pull request to fix this bug?**

Not at this time.

**Comment by mdjermanovic** [comment]
Hi @apples, thanks for the bug report!

Confirmed in our [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhbWVsY2FzZTogW1wiZXJyb3JcIl0gKi9cblxuY29uc3Qgb2JqID0ge307XG5cbmNvbnN0IHsgWydmb28nXTogX2ZvbyB9ID0gb2JqO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==), I'm working on this.
